icu4c/source/common/ubidiln.cpp

```diff
@@ -620,7 +620,7 @@ ubidi_getRuns(UBiDi *pBiDi, UErrorCode*) {
 
             /* search for the run limits and initialize visualLimit values with the run lengths */
             i=0;
-            do {
+            while(i<limit) {
```

Comment:
line 578 says that `limit>0`. let's assert that if necessary, rather than change the logic.
after 26 years with this code as is, i doubt there is a real problem to be fixed.